cmd/root.go

```diff
@@ -50,6 +51,25 @@ func init() {
 		}
 	} else {
 		fmt.Fprintf(os.Stderr, "Using configuration file: %s\n", viper.ConfigFileUsed())
+
+	}
+	validateConfiguration()
+}
+
+func validateConfiguration() {
+	u, _ := url.Parse(viper.GetString(argRootServer))
+	if u.Scheme == "" {
+		viper.Set(argRootServer, "https://"+viper.GetString(argRootServer))
+		u, _ = url.Parse(viper.GetString(argRootServer))
+		log.Info("Protocol is not specified, HTTPS is used by default.")
+	}
+	if u.Scheme == "http" {
+		viper.Set(argRootServer, "https://"+u.Host)
+		log.Info("HTTPS is forcely used instead of HTTP.")
+	}
```

Comment:
Same here, there are scenarios where you want to use http and forcing https is just "hiding" a misconfiguration in the mender-cli settings.